packages/kbn-openapi-generator/docs/USAGE_GUIDE.md

```diff
@@ -0,0 +1,432 @@
+## OpenAPI Generator (`kbn-openapi-generator`) Usage Guide
+
+OpenAPI code generator (`kbn-openapi-generator`) could be used to generate runtime types, documentation, server stub implementations, clients, and much more (later just artifacts) given OpenAPI specification. Usually the main goal is to generate Zod schemas and related TypeScript types and integrate them into routes implementation and the other code. This guide focuses on mainly on that topic.
+
+Any OpenAPI spec files should conform to [OpenAPI Specification 3.0](https://swagger.io/specification/v3/). OpenAPI Generator supports a subset of the specification.
+
+### How to organize files with OpenAPI specs?
+
+You have quite a lot of freedom on organizing your OpenAPI specs and code generator supports any valid OpenAPI. Though we recommend to follow rules derived from using OpenAPI specs and code generation in Rule Management team
```

Comment:
I'd consider reversing the order of the sentences here to encourage people more strongly to follow the recommendations. We (you) have put a lot of thought into how to structure these schemas, so I think it makes sense to say basically "follow our example if possible because it'll save us all some headaches, but there's flexibility if you need it". Putting the flexibility first makes it sound like they should use the flexibility even if they don't need to IMO.